proxy/src/main/java/com/velocitypowered/proxy/plugin/VelocityEventManager.java

```diff
@@ -129,28 +142,35 @@ public boolean shouldRegister(@NonNull Object listener, @NonNull Method method)
         }
 
         @Override
-        public @NonNull PostOrder postOrder(@NonNull Object listener, @NonNull Method method) {
-            return PostOrder.NORMAL; // TODO: Allow customizing his
+        public int postOrder(@NonNull Object listener, @NonNull Method method) {
+            return method.getAnnotation(Subscribe.class).order().ordinal();
         }
 
         @Override
         public boolean consumeCancelledEvents(@NonNull Object listener, @NonNull Method method) {
-            return true;
+            return !method.getAnnotation(Subscribe.class).ignoreCancelled();
```

Comment:
Velocity's "cancelled event" is based on `ResultedEvent<T>`. As this presently stands, all "cancelled events" will still be called by the event system.

Reply:
Ahh, my bad :)